### PR TITLE
[RFC] Bubble serialization up

### PR DESCRIPTION
**Pull request by andreaTP**
I applied this change just to `action` since is the most tested and shows the intention.

Here we:
 - remove the proto logic in the `Component`
 - inject the serializer (still through the `Component` to keep the constructor signature ... for now)
 - use the injected `serializationSupport` for everything
 
It's still not a "final design", but a step forward, and we need to do more cleanup and write more tests before applying the same pattern to the other components.

### CODE REVIEW

**Comment by retgits** [comment]
For a user of the SDK, will this change anything in the code they write?

**Comment by andreaTP** [comment]
> For a user of the SDK, will this change anything in the code they write?

No

**Review by pvlugter — COMMENTED** [review]
LGTM to move all the serialization support together.

Failing the integration testkit test.

File: sdk/src/serialization-support.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  getEffectSerializer() {
+    if (!this.allComponents) {
+      throw new Error('Please run `setComponents` first');
+    } else {
+      return new EffectSerializer(this.allComponents);
+    }
+  }
+
+  getGrpc() {
```
Comment (andreaTP) [inline sdk/src/serialization-support.js:53]: Having to expose `Grpc` and `Service` doesn't sounds great 🙁 
Probably this PR needs a few more steps to go through ...

**Comment by johanandren** [comment]
Lets' close this for now and possibly revisit in the future

**Pull request closed without merging**